src/reader.rs

```diff
@@ -5116,29 +5108,29 @@
             if search_term != zelf.pager.search_field_line.text() {
                 return; // Stale request.
             }
+            let history_len = zelf.history.size();
             let history_pager = zelf.history_pager.as_mut().unwrap();
-            history_pager.direction = direction;
-            match direction {
-                SearchDirection::Forward => {
-                    history_pager.can_go_backwards = true;
-                    if index == 0 || index <= result.final_index {
-                        return;
-                    }
-                    history_pager.history_index_start = result.final_index;
-                    history_pager.history_index_end = index;
-                }
-                SearchDirection::Backward => {
-                    history_pager.history_index_start = index;
-                    history_pager.history_index_end = result.final_index;
-                    history_pager.can_go_backwards = result.have_more_results;
-                }
-            };
-            zelf.pager.extra_progress_text = if result.have_more_results {
-                wgettext!("Search again for more results")
-            } else {
-                L!("")
+
+            if result.matched_commands.is_empty() && result.range != (0..history_len + 1) {
+                history_pager.start = history_len;
+                zelf.fill_history_pager(HistoryPagerInvocation::Advance, SearchDirection::Forward);
+                return;
```

Comment:
This "recursive" call is sus;
It's for the the scenario where we delete the last element on the last page, right.
It might be better to do this only for that invocation (in HistoryPagerDelete),
and maybe earlier, inside the performer.

I guess it's at odds with the current approach  to `HistoryPagerInvocation::Refresh` when at the last page.
Each deletion makes the page smaller by one. Maybe we should change that so the page size always stays the same (until there are not enough items in history left)

But I think it's also okay to leave it like this for now, and maybe change it later.